### ISSUE TITLE
Pods running on master doesn't have correct pod ip

### ISSUE THREAD

**Issue by kzwang** [kubernetes/kops#448]
If the pod is running on the master (deployed using DaemonSet), it doesn't have a correct pod IP address and not able to access the internet.

From the log, it looks like kubenet is configuring CIDR to the fake one `10.123.45.0/30` and ignoring the actual CIDR assigned by the controller manager.

```
I0915 05:00:06.619218    1702 kubelet.go:3907] Setting Pod CIDR:  -> 10.123.45.0/30
I0915 05:00:06.619227    1702 kubenet_linux.go:218] PodCIDR is set to "10.123.45.0/30"
I0915 05:00:06.619238    1702 kubenet_linux.go:225] CNI network config set to {
  "cniVersion": "0.1.0",
  "name": "kubenet",
  "type": "bridge",
  "bridge": "cbr0",
  "mtu": 1500,
  "addIf": "eth0",
  "isGateway": true,
  "ipMasq": false,
  "ipam": {
    "type": "host-local",
    "subnet": "10.123.45.0/30",
    "gateway": "10.123.45.1",
    "routes": [
      { "dst": "0.0.0.0/0" }
    ]
  }
}



I0915 05:00:06.801418    1702 kubelet.go:3907] Setting Pod CIDR: 10.123.45.0/30 -> 100.96.1.0/24
I0915 05:00:06.801468    1702 kubenet_linux.go:214] Ignoring subsequent pod CIDR update to 100.96.1.0/24
```

The IP address assigned to the pod is `10.123.45.2`

**Comment by justinsb** [comment]
Thanks for reporting - will try to reproduce!

**Comment by shamil** [comment]
Since updating to 1.4.1, I started to get this as well. All PODs started on master have wrong IPs

**Comment by shamil** [comment]
@justinsb, seems like it only affects if the networking was set to `kubenet`. btw, I'm still running k8s v1.3.7. Is it even supported with kops 1.4.1?

**Comment by shamil** [comment]
@justinsb did you manage to reproduce this?

**Comment by chrislovecnm** [comment]
We support 1.4.4 and with the TLS issue please upgrade :)

**Comment by justinsb** [comment]
This will be fixed in 1.5 by https://github.com/kubernetes/kubernetes/pull/35526

If you use weave you should side-step the problem (though you may step into different ones... I'm not sure!)

Your pod should still be able to access the internet though .. it just can't access the pod network.  You need to make sure `dnspolicy` is set to `default`, not `clusterfirst` IIRC, because otherwise it will try to talk to the cluster DNS service, which is on the pod network and it can't reach.

Until the fix is merged though, there is no way for a pod on the master to talk to the pod network with kubenet.

**Comment by fwynyk** [comment]
I had this same problem, and find that forcing the pods to use real IP addresses of your host network solves the problem. To do that, you can put this in your DaemonSet configuration, under 'template' statement:

spec:
     hostNetwork: true

Hope to get this fixed on 1.5.

**Comment by chrislovecnm** [comment]
@fwynyk / @justinsb you know if we have a bug in core?

**Comment by justinsb** [comment]
With #1294, a daemonset pod on the master (without hostNetwork) got `IP: 100.96.1.2`, and was able to reach https://kubernetes.default (for example)

**Comment by justinsb** [comment]
i.e. this will be fixed in kops 1.5 :-)

**Comment by justinsb** [comment]
I'm going to close as I believe this is fixed in e.g. kops 1.5.0 beta1 which is available for download.

Please comment/reopen though if you encounter the issue again in kops 1.5.0 beta1 or later, when running with kubernetes version >= 1.5.0
